### DISCUSSION TITLE
No mining of firo on 2miners

### DISCUSSION THREAD

**Discussion by DeBelsch** [RainbowMiner/RainbowMiner#1684] [Q&A] [answered]
Hello,

I setup rainbowminer today and i love it, but somehow it won't benchmark or mine firo on 2miners. I have setup the wallet and it sees what is in the wallet but won't use it. When i select only MTP for algo, it says: no pool available.

Thank you for yout work.

**Comment by RainbowMiner** [comment]
Could you please create a Debug file and upload it here? Just open `http://localhost:4000` and click "Debug file" on the left hand side (scroll down a bit). It will create a zip file with all sensitive data stripped.

**Comment by DeBelsch** [comment]
Thank you.

[debug_2021-04-27.zip](https://github.com/RainbowMiner/RainbowMiner/files/6385606/debug_2021-04-27.zip)

**Comment by RainbowMiner** [comment, marked as answer]
Ok, so that's easy to solve. You need to tell the 2Miners module, which FIRO wallet to use. To use the coin setup in coins.config.txt, use `$FIRO` in pools.config.txt's 2Miners section. This works for every other coins in coins.config.txt as well.

Like that:
```
  "FIRO":  "$FIRO",
```

Short explanation: every pool module needs to know, which wallet address(es) are to be used. You can either insert the full wallet address, or use the variable notation with the `$` sign to refer to all the coins in coins.config.txt (like in the example above).

**Comment by DeBelsch** [comment]
Thanks. It show me now the balance of the firo address on 2miners, but i still can't mine it. Maybe i have understood something wrong? Iedited the file like to this.
`{
    "2Miners":  {
                    "FIRO":  "$FIRO",
                    "FIRO-Params":  "",
                    "Worker":  "$WorkerName",
                    "Penalty":  "0",
                    "DataWindow":  "",
                    "StatAverage":  "",
                    "StatAverageStable":  "",
                    "MaxMarginOfError":  "100",
                    "SwitchingHysteresis":  "",
                    "MaxAllowedLuck":  "",
                    "MaxTimeSinceLastBlock":  "",
                    "Region":  "",
                    "SSL":  "",
                    "ETH":  "$ETH",
                    "ETH-Params":  "",
                    "Algorithm":  "",
                    "ExcludeAlgorithm":  "",
                    "CoinName":  "",
                    "ExcludeCoin":  "",
                    "CoinSymbol":  "",
                    "ExcludeCoinSymbol":  "",
                    "CoinSymbolPBM":  "",
                    "MinerName":  "",
                    "ExcludeMinerName":  "",
                    "EnableAutoCoin":  "1",
                    "EnablePostBlockMining":  "0",
                    "FocusWallet":  "",
                    "AllowZero":  "0"
                },`

**Comment by RainbowMiner** [comment]
Oh! Good, that you have reported this. It seems like 2Miners have switched their RPC url for FIRO mining. I have fixed the pool modules.

**Comment by DeBelsch** [comment]
Thanks :)
